src/megaapi_impl.cpp

```diff
@@ -20649,6 +20670,38 @@ void MegaApiImpl::sendPendingRequests()
 
             if (!flag)
             {
+                long long cancelledPending = 0;
+                // 1. cancel queued transfers, not yet started (and not added to cache), up to the last one queued when cancelTransfers was queued
+                auto lastQueuedTransfer = request->getTransferTag();
+                while (MegaTransferPrivate *transfer = transferQueue.pop())
+                {
+                    if (transfer->getPlaceInQueue() > lastQueuedTransfer)
+                    {
+                        break;
```

Comment:
I think we will have popped off one too many transfers off the queue here - should we use push_front to put it back?

Reply:
you are right, further more: we are poping sync transfers & those with different direction too! fixing it, by having a method that pops up exactly what we want in TransferQueue. thanks for spotting that